src/org/traccar/Context.java

```diff
@@ -297,39 +333,8 @@ public static void init(String[] arguments) throws Exception {
 
         identityManager = deviceManager;
 
-        if (config.getBoolean("geocoder.enable")) {
-            String type = config.getString("geocoder.type", "google");
-            String url = config.getString("geocoder.url");
-            String key = config.getString("geocoder.key");
-            String language = config.getString("geocoder.language");
-
-            int cacheSize = config.getInteger("geocoder.cacheSize");
-            switch (type) {
-                case "nominatim":
-                    geocoder = new NominatimGeocoder(url, key, language, cacheSize);
-                    break;
-                case "gisgraphy":
-                    geocoder = new GisgraphyGeocoder(url, cacheSize);
-                    break;
-                case "mapquest":
-                    geocoder = new MapQuestGeocoder(url, key, cacheSize);
-                    break;
-                case "opencage":
-                    geocoder = new OpenCageGeocoder(url, key, cacheSize);
-                    break;
-                case "bingmaps":
-                    geocoder = new BingMapsGeocoder(url, key, cacheSize);
-                    break;
-                case "factual":
-                    geocoder = new FactualGeocoder(url, key, cacheSize);
-                    break;
-                case "geocodefarm":
-                    geocoder = new GeocodeFarmGeocoder(key, language, cacheSize);
-                    break;
-                default:
-                    geocoder = new GoogleGeocoder(key, language, cacheSize);
-                    break;
-            }
+        if (config.getBoolean("geocoder.enable") || config.getBoolean("report.retryGeocoding")) {
```

Comment:
I think `retryGeocoding` is not a very good name. Also, there is no way to disable geocoding for each location. Maybe instead of `retryGeocoding` we can add a parameter `geocoder.lazy` or something like that?

Reply:
`geocoder.enable` = `false` disables geocoder for each position.
May be `geocoder.onDemand` ?

Reply:
It's a bit confusing if with `geocoder.enable` you can still get geocoding results. Maybe `geocoder.onRequest`?

Reply:
Agree, if we were not concern of backward compatibility we could make like this
`geocoder.enable` - main switch
`geocoder.everyPosition` - enables Geocoder pipeline handler
`geocoder.onRequest` - enables geocoding for trips/stops

Reply:
I think we don't need 3 options. We just need two:
`geocoder.enable` - main switch
`geocoder.onRequest` - disable geocoding for every position and enable for reports

Reply:
I want to cover a case when geocoding in every position was enabled, but address of endpoints is empty by some reason (network or invalid position, or limit reached)

Reply:
Then i suggest:
`geocoder.enable` - main switch
`geocoder.ignorePositions` - disable automated geocoding for every position; disabled by default because `false` is default value
`geocoder.onRequest` - enable geocoding by request